### ISSUE TITLE
add_field and tags options fail with s3 input and multiline codec

### ISSUE THREAD

**Issue by ogarcialop** [elastic/logstash#9686]
- Version: 6.2.4
- Environment: Official docker image for logstash 6.2.4

```
input {
  s3 {
    codec => multiline {
      pattern => "%{TIMESTAMP_ISO8601}"
      what => "previous"
      negate => true
    }
    "access_key_id" => "<key_id>"
    "secret_access_key" => "<access_key>"
    "bucket" => "<my_bucket>"
    "exclude_pattern" => "log$|\d$" 
    "temporary_directory" => "/tmp/logstash/elk_input_0"
    tags => ["i-03b54a7132997994d"]
    add_field => {
      "[@metadata][version]" => "6.2.4"
      "beat.hostname" => "${HOSTNAME}"
      "beat.version" => "6.2.4"
      "beat.name" => "logstash-s3-input"
      "instance.id" => "i-03b54a"
      "instance.description" => ""
      "instance.name" => "i-03b54a"
      "prospector.module" => "elk"
      "prospector.name" => "elk"
    } 
    "additional_settings" => {
    "force_path_style" => true
    "follow_redirects" => false
    }
  }
}

output {
  elasticsearch {
    hosts => ["es:9200"]
    index => "elk-fb.%{[beat.version]}-%{[instance.id]}-s3-%{+yyyy.MM.dd}"
    pipeline => "elk-fb.%{[beat.version]}"
  }
}
      
```

Logstash output:
```
[2018-05-30T08:39:00,379][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", :_type=>"doc", :_routing=>nil, :pipeline=>"elk-fb.%{[beat.version]}"}, #<LogStash::Event:0x64982bc3>], :response=>{"index"=>{"_index"=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", "_type"=>"doc", "_id"=>nil, "status"=>400, "error"=>{"type"=>"illegal_argument_exception", "reason"=>"pipeline with id [elk-fb.%{[beat.version]}] does not exist"}}}}
[2018-05-30T08:39:00,381][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", :_type=>"doc", :_routing=>nil, :pipeline=>"elk-fb.%{[beat.version]}"}, #<LogStash::Event:0x3c04b5be>], :response=>{"index"=>{"_index"=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", "_type"=>"doc", "_id"=>nil, "status"=>400, "error"=>{"type"=>"illegal_argument_exception", "reason"=>"pipeline with id [elk-fb.%{[beat.version]}] does not exist"}}}}
[2018-05-30T08:39:01,030][WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. {:status=>400, :action=>["index", {:_id=>nil, :_index=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", :_type=>"doc", :_routing=>nil, :pipeline=>"elk-fb.%{[beat.version]}"}, #<LogStash::Event:0x314258ce>], :response=>{"index"=>{"_index"=>"elk-fb.%{[beat.version]}-%{[instance.id]}-s3-2018.05.30", "_type"=>"doc", "_id"=>nil, "status"=>400, "error"=>{"type"=>"illegal_argument_exception", "reason"=>"pipeline with id [elk-fb.%{[beat.version]}] does not exist"}}}}


```

**Comment by ogarcialop** [comment]
Still present in version 6.3.0

**Comment by jinliangXX** [comment]
Still present in version 6.7.0

When s3 and multiline are used at the same time, the metadata information([@metadata]) cannot be obtained.

**Comment by TheVastyDeep** [comment]
Correct. See issue [153](https://github.com/logstash-plugins/logstash-input-s3/issues/153) on the S3 input.

**Comment by jgough** [comment]
There seem to be at least 3 pull requests to fix this issue (including one I have just submitted) but none have seem to have been approved.

https://github.com/logstash-plugins/logstash-input-s3/pull/218
https://github.com/logstash-plugins/logstash-input-s3/pull/190
https://github.com/logstash-plugins/logstash-input-s3/pull/173

**Comment by kaisecheng** [comment]
The issue is fixed in `3.8.3`
